astropy/table/tests/test_table.py

```diff
@@ -2589,6 +2589,27 @@ def test_table_meta_copy_with_meta_arg():
     assert t.meta[1] is not meta[1]
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.array([1], dtype=np.int32),
```

Comment:
Here too, I'd parametrize just some examples of `object`.

p.s. The test itself would be independent of my suggested change.